java-vtl-script/src/main/java/no/ssb/vtl/script/operations/join/InnerJoinOperation.java

```diff
@@ -19,50 +19,40 @@
  * limitations under the License.
  * =========================LICENSE_END==================================
  */
-/*-
- * #%L
- * java-vtl-script
- * %%
- * Copyright (C) 2016 Hadrien Kohl
- * %%
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- * #L%
- */
 
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
 import com.google.common.collect.Table;
 import no.ssb.vtl.model.Component;
 import no.ssb.vtl.model.DataPoint;
 import no.ssb.vtl.model.DataStructure;
 import no.ssb.vtl.model.Dataset;
-import no.ssb.vtl.model.VTLObject;
+import no.ssb.vtl.model.Order;
+import no.ssb.vtl.script.support.Closer;
 
+import java.io.IOException;
 import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
 import java.util.function.BiFunction;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
 
-/**
- * Represent an inner join on datasets.
- */
 public class InnerJoinOperation extends AbstractJoinOperation {
 
+
     public InnerJoinOperation(Map<String, Dataset> namedDatasets) {
-        super(namedDatasets, Collections.emptySet());
+        this(namedDatasets, Collections.emptySet());
     }
 
     public InnerJoinOperation(Map<String, Dataset> namedDatasets, Set<Component> identifiers) {
-        super(namedDatasets, identifiers);
+        super(namedDatasets.entrySet()
+                .stream()
+                .sorted(Comparator.comparing(stringDatasetEntry -> stringDatasetEntry.getValue().getSize().orElse(Long.MAX_VALUE))).collect(ImmutableMap.toImmutableMap(Map.Entry::getKey, Map.Entry::getValue)), identifiers);
```

Comment:
Should we remove this to have the possibility to change the order in the expression? That is, to pass it on in the same order as written in the expression?

Reply:
Right, I pushed a fix.